cmd/nerdctl/run_mount_linux_test.go

```diff
@@ -340,6 +343,55 @@ func TestRunBindMountBind(t *testing.T) {
 	base.Cmd("exec", containerName, "grep", "/mnt2", "/proc/mounts").AssertOutWithFunc(f("ro"))
 }
 
+func TestRunBindMountNonRecursive(t *testing.T) {
+	if rootlessutil.IsRootless() {
+		t.Skip("must be superuser to use mount")
+	}
+	t.Parallel()
+	base := testutil.NewBase(t)
+
+	tmpDir1, err := os.MkdirTemp(t.TempDir(), "rw")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(tmpDir1)
+	tmpDir1Mnt := filepath.Join(tmpDir1, "mnt")
+	if err := os.MkdirAll(tmpDir1Mnt, 0700); err != nil {
+		t.Fatal(err)
+	}
+
+	tmpDir2, err := os.MkdirTemp(t.TempDir(), "ro")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(tmpDir2)
+
+	if err := mobymount.Mount(tmpDir2, tmpDir1Mnt, "none", "bind,ro"); err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		if err := mobymount.Unmount(tmpDir1Mnt); err != nil {
+			t.Fatal(err)
+		}
+	}()
+
+	base.Cmd("run",
+		"--rm",
+		"--mount", fmt.Sprintf("type=bind,bind-nonrecursive,src=%s,target=/mnt1", tmpDir1),
+		testutil.AlpineImage,
+		"sh", "-euxc", "apk add findmnt -q && findmnt -nR /mnt1",
```

Comment:
nit: Could we have tests for `-v` + `bind`/`rbind`?

Reply:
SGTM, fixed